CheckStaticAnalysisQualityGate/task.json

```diff
@@ -16,22 +16,59 @@
     "Patch": 4
   },
   "minimumAgentVersion": "2.144.0",
-  "groups": [
-    {
-      "name": "qualityGateOptions",
-      "displayName": "Quality Gate options",
-      "isExpanded": false
-    }
-  ],
   "inputs": [
     {
-      "name": "inputField",
+      "name": "type",
+      "type": "radio",
+      "label": "Type",
+      "defaultValue": "total",
+      "required": true,
+      "helpMarkDown": "You can choose which property will be used to compare the provided threshold to issues of a given severity.",
+      "options": {
+        "total": "Total issues",
+        "new": "New issues"
+      }
+    },
+    {
+      "name": "severity",
+      "type": "radio",
+      "label": "Severity",
+      "defaultValue": "error",
+      "required": true,
+      "helpMarkDown": "Severity of issues.",
+      "options": {
+        "error": "Error",
+        "warning": "Warning",
+        "note": "Note"
+      }
+    },
+    {
+      "name": "threshold",
+      "type": "string",
+      "label": "Threshold",
+      "defaultValue": "0.0",
+      "required": true,
+      "helpMarkDown": "If the actual number of issues is greater or equal than this threshold then a build is considered as unstable or failed, respectively."
+    },
+    {
+      "name": "referenceBuild",
       "type": "string",
-      "label": "Quality Gate",
+      "label": "Reference build",
       "defaultValue": "",
       "required": false,
-      "groupName": "qualityGateOptions",
-      "helpMarkDown": "Set Quality Gate."
+      "helpMarkDown": "Specify reference build number. Do not specify a build number, if you want to use the latest success build with Static Analysis report."
+    },
+    {
+      "name": "buildStatus",
+      "type": "radio",
+      "label": "Build status if quality gate fails",
+      "defaultValue": "failed",
+      "required": true,
+      "helpMarkDown": "Determines if the build should be set to unstable or failed if the quality gate is not passed.",
```

Comment:
How about: Mark the build as unstable or failed if the quality gate is not passed.